site/en/docs/extensions/mv3/getstarted/tut-focus-mode/index.md

```diff
@@ -0,0 +1,374 @@
+[here][github-focus-mode-icons].
+
+### Step 2: Initialize the extension {: #step-2 }
+
+Extensions monitor browser events using the [extension service worker][doc-sw]. Service workers have
+a special JavaScript environment; they load to handle events, and the browser terminates a service
+worker when it's no longer needed.
+
+Start by adding the following code to register the service worker in the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "background": {
+      "service_worker": "background.js"
+  },
+  ...
+}
+```
+
+The first event the service worker will listen for is
+[`runtime.onInstalled()`][runtime-oninstalled]. This method allows the extension to set an initial
+state or complete some tasks on installation. Since this example will only handle two states (ON and
+OFF), it will track each tab's state through the _extension badge_.
+
+{% Aside 'key-term' %}
+
+The [extension badge][action-badge] is a colored banner on top of the extension action (toolbar
+icon).
+
+{% endAside %}
+
+Create a file called `background.js` and add the following code to set the initial state to “OFF”:
+
+{% Label %}background.js:{% endLabel %}
+
+```js
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.action.setBadgeText({
+    text: "OFF",
+  });
+});
+...
+```
+
+{% Details %}
+{% DetailsSummary %}
+💡 **What if I have a more complex state?**
+{% endDetailsSummary %}
+
+You can use the [Storage API][api-storage] to store state data.
+
+{% endDetails %}
+
+### Step 3: Enable the extension action {: #step-3 }
+
+The _extension action_ controls the extension’s toolbar icon. So whenever the user clicks on the
+extension action, it will either run some code (like in this example) or display a popup. Add the
+following code to declare the extension action in the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "action": {
+    "default_icon": {
+      "16": "images/icon-16.png",
+      "32": "images/icon-32.png",
+      "48": "images/icon-48.png",
+      "128": "images/icon-128.png"
+    }
+  },
+  ...
+}
+```
+
+#### Use the activeTab permission to protect user privacy {: #active-tab}
+
+The [`activeTab`][doc-active-tab] permission grants the extension _temporary_ ability to execute code
+on the currently active tab. It also allows access to [sensitive properties][active-tab-allows] of
+the current tab.
+
+This permission is enabled when the user **_invokes_** the extension. In this case, the user invokes the
+extension by clicking on the extension action.
+
+{% Details %}
+{% DetailsSummary %}
+💡 **What other user interactions enable the activeTab permission?**
+{% endDetailsSummary %}
+
+Other user gestures that enable the activeTab permission are the following:
+
+- Pressing a keyboard shortcut combination.
+- Selecting a context menu item.
+- Accepting a suggestion from the omnibox.
+- Opening an extension popup.
```

Comment:
At the moment the framing of this list implies that any time the user performs one of these actions, activeTab will be granted to any interested extension. On my first pass through this section, I found myself revising each line to include a qualification like "Pressing a keyboard shortcut combination _registered by the extension_." If possible, we should work this into the paragraph before the list.

Also, "Opening an extension popup" is too narrow as the extension may not have a popup associated with it's action. Triggering the action – either by click or keyboard shortcut – will gant activeTab.

Reply:
Changed copy 💯